lazer/sdk/js/src/socket/websocket-pool.ts

```diff
@@ -144,11 +163,14 @@ export class WebSocketPool {
    * Handles incoming websocket messages by deduplicating identical messages received across
    * multiple connections before forwarding to registered handlers
    */
-  dedupeHandler = (data: WebSocket.Data): void => {
-    const cacheKey =
-      typeof data === "string"
-        ? data
-        : Buffer.from(data as Buffer).toString("hex");
+  dedupeHandler = async (data: WebSocket.Data): Promise<void> => {
+    let cacheKey = "";
+    if (typeof data === "string") {
+      cacheKey = data;
+    } else {
+      const buff = await IsomorphicBuffer.fromWebsocketData(data);
+      cacheKey = buff.toString("hex");
+    }
```

Comment:
Similar nit -- this can be trivially written without using a mutable binding and I strongly prefer to avoid unneeded mutable bindings when not strictly required:

```ts
const cacheKey = typeof data === "string" ? data : (
  (await IsomorphicBuffer.fromWebsocketData(data)).toString("hex")
);
```

If you don't like the ternary (I don't love it either), you can extract to a `getCacheKey` function too, which avoids the mutable binding AND decomposes the function a bit too which is nice.

Reply:
done in 7aebe1f1b2c2709ca9de50a2e0b09efc8496d4b5